File: cpp/alerts/block_timeout.cpp
#include "./block_timeout.h"

namespace ltrs {

int block_timeout_alert_get_block_index(lt::block_timeout_alert* alert) {
    return alert->block_index;
}

PieceIndex block_timeout_alert_get_piece_index(lt::block_timeout_alert* alert) {
    return PieceIndex { .inner = static_cast<int32_t>(alert->piece_index) };
}

} // namespace ltrs

File: src/ffi/alerts/block_timeout.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type block_timeout_alert = crate::ffi::ffi::block_timeout_alert;
    }

    unsafe extern "C++" {
        include!("cpp/alerts/block_timeout.h");
        type PieceIndex = crate::ffi::ffi::PieceIndex;
        unsafe fn block_timeout_alert_get_block_index(alert: *mut block_timeout_alert) -> i32;
        unsafe fn block_timeout_alert_get_piece_index(
            alert: *mut block_timeout_alert,
        ) -> PieceIndex;
    }
}
